src/v/iceberg/transaction.h

```diff
@@ -40,6 +40,10 @@ class transaction {
       chunked_vector<data_file>,
       chunked_vector<std::pair<ss::sstring, ss::sstring>> snapshot_props = {});
 
+    // Removes expired snapshots from the table. Note, this does not perform IO
+    // to delete any now-orphaned metadata or data.
+    ss::future<txn_outcome> remove_expired_snapshots(model::timestamp);
```

Comment:
So I peeked into the next PR and it seems that to determine the list of expired snapshots we have to do a rather unpleasant load-before, load-after sequence. I wonder if the transaction could return the set of expired snapshots directly.

Reply:
It could, I had that implemented initially too, but I ended up going with this because this is what the Java implementation does. The theoretical concern is that the catalog may return a different set of snapshots to remove, or may end up removing more than just what this transaction computed. It seems debatable whether Redpanda should honor its own removals vs those given by the catalog, and I opted to give more weight to the catalog. Happy to discuss further if you feel strongly about this.

Reply:
No, not feeling particularly strongly, it makes sense to follow what the reference impl does.